src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -43,19 +46,170 @@ const SKUpdate& SKNMEA2000Parser::parse(const SKSourceInput& input, const tN2kMs
+// Offset to millis() for recalculation of true date/time
+// TODO: after 42 days millis() overflow, new time sync needed
+const SKUpdate& SKNMEA2000Parser::parse126992(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;          // Sequence ID
+  uint16_t SystemDate;        // Days since 1970-01-01
+  double SystemTime;          // seconds since midnight mit 2 Kommastellen
+  tN2kTimeSource TimeSource;  // see tN2kTimeSource (z.B.: GPS)
+
+    if (ParseN2kSystemTime(msg,sid,SystemDate,SystemTime,TimeSource) ) {
+
+      /*
+      // static global variables needed....
+      if ( SystemDate > 0 && SystemTime > 0 && dateTime::timeStampNeeded ) {
+        // time function class
+        dt.setTimeStamp( &SystemDate, &SystemTime);
+        // the first 20 Seconds try to get date and time from N2k bus
+        if ( millis() > 20000 ) dateTime::timeStampNeeded = false;
+      }
+      */
+    } else {
+      INFO("Unable to parse NMEA2000 with PGN %i", msg.PGN);
+    }
+  // nothing todo at the moment
+  return _invalidSku;
+}
+
+//  ***********************************************
+//   PGN 127245 Rudder
+//        →  sid
+//        →  RudderPosition [rad]
+// *  ********************************************** */
+const SKUpdate& SKNMEA2000Parser::parse127245(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char Instance;
+  tN2kRudderDirectionOrder RudderDirectionOrder;
+  double RudderPosition;
+  double AngleOrder;
+
+  if (ParseN2kRudder(msg,RudderPosition,Instance,RudderDirectionOrder,AngleOrder) ) {
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+    update->setSteeringRudderAngle(RudderPosition);
+    DEBUG("Rudder: %d",SKRadToDeg(RudderPosition));
+    _sku = update;
+    return *_sku;
   }
-  // Water Depth
-  if (msg.PGN == 128267) {
-    return parse128267(input, msg, timestamp);
+  else {
+    INFO("Unable to parse NMEA2000 with PGN %i", msg.PGN);
   }
+  // nothing todo at the moment....
+  return _invalidSku;
+}
+
+// *****************************************************************************
+//   PGN 127250 VESSEL HEADING RAPID
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse127250(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  tN2kHeadingReference HeadingReference;
+  double Heading = 361; // not valid value
+  double Deviation = 0; // NAN would be better!
+  double Variation = 0; // NAN would be better!
 
-  DEBUG("No known conversion for PGN %i", msg.PGN);
+  if (ParseN2kHeading(msg,sid,Heading,Deviation,Variation,HeadingReference) ) {
+
+    if ( String(HeadingReference) == "magnetic" ) {
```

Comment:
You can just do `headingReference == N2khr_magnetic` (it is defined [here](https://github.com/ttlappalainen/NMEA2000/blob/master/src/N2kMessages.h#L60)).

I think changing the `heading` value based on it's type is wrong. What we should do is set a different signalK variable.

If it's magnetic, we set: `headingMagnetic` and `magneticVariation` (if the variation is set). SignalK does not support deviation now but I think it should be added. I will ask them why it's not here now.

If it's true, we set `headingTrue`.

Reply:
I'll do all you mentioned here and above.
Put the code for deviation commented to be ready when Signal K is.
(I know, that I should not put commented code... :smirk:)

Could you please ask them for support of reference type of speed through water too, as we will have to correct readings from paddle wheel by the cosinus of leeway later on.
(Ultra sound and doppler sound do not need this correction)
**Reference types:**
//                Paddle_wheel=0,
//                Pitot_tube=1,
//                Doppler_log=2,
//                Ultra_Sound=3,
//                Electro_magnetic=4

Reply:
I'll do all you mentioned here and above.
Put the code for deviation commented to be ready when Signal K is.
(I know, that I should not put commented code... :smirk:)

Could you please ask them for support of reference type of speed through water too, as we will have to correct readings from paddle wheel by the cosinus of leeway later on.
(Ultra sound and doppler sound do not need this correction)
**Reference types:**
//                Paddle_wheel=0,
//                Pitot_tube=1,
//                Doppler_log=2,
//                Ultra_Sound=3,
//                Electro_magnetic=4

Reply:
I have asked in the #specification channel on the SignalK slack group. You should join. Unless they have a reason to not include it, I guess the best thing to do would be to submit a pull-request to them ;)